lib/dialog.rb

```diff
@@ -1,7 +1,6 @@
 class Dialog
 
-	def initialize(game_state, args = {})
-		@game_state = game_state
+	def initialize(args = {})
 		@output = args[:output] ||= ->(string) { puts string }
```

Comment:
passing procs/lambdas around is a perfectly viable solution, but might not be considered "idiomatic" Ruby. I would instead consider passing an object that has a method that takes a string and prints it to the console